pkg/bot/slack.go

```diff
@@ -93,74 +96,68 @@ func NewSlack(log logrus.FieldLogger, cfg config.Slack, executorFactory Executor
 	}, nil
 }
 
-// Start starts the Slack RTM connection and listens for messages
+// Start starts the Slack Websocket connection and listens for messages
 func (b *Slack) Start(ctx context.Context) error {
 	b.log.Info("Starting bot")
 
-	rtm := b.client.NewRTM()
+	websocketClient := socketmode.New(b.client)
+
 	go func() {
 		defer analytics.ReportPanicIfOccurs(b.log, b.reporter)
-		rtm.ManageConnection()
+		socketRunErr := websocketClient.Run()
+		if socketRunErr != nil {
+			reportErr := b.reporter.ReportFatalError(socketRunErr)
+			if reportErr != nil {
+				b.log.Errorf("while reporting socket error: %s", reportErr.Error())
+			}
+		}
 	}()
 
 	for {
 		select {
 		case <-ctx.Done():
 			b.log.Info("Shutdown requested. Finishing...")
-			return rtm.Disconnect()
```

Comment:
Had a quick look around to understand how disconnects work in the context of websockets.

For my understanding only ... to confirm: 
1 - [disconnects are not predictable](https://api.slack.com/apis/connections/socket-implement#disconnect) but they are handled by the Slack library. 
2 - When we terminate BotKube we're assuming the server will terminate the connection?

Reply:
Really good catch, please see my updated impl. I have used RunContext(ctx) instead of Run(), so that once ctx is done, [it will also close connection on websocket.](https://github.com/slack-go/slack/blob/06ab08cc28df2814027aae2cd3b1a65e87035bc9/socketmode/socket_mode_managed_conn.go#L147)

Reply:
Easy to miss. I was really trying to make sense of it myself having never used the library.